plugin/inc/installsteps/addcustomfieldsinstallstep.class.php

```diff
@@ -175,16 +175,16 @@ public static function getColumnsSql($fieldsData): array
 
             switch (true) {
             case $field_type === 'yesno':
-                $fields[$field_name] = "tinyint $mandatory" . ($default !== null ? " default $default" : '');
+                $fields[$field_name] = "tinyint $mandatory" . ($default === null && $mandatory == '' ? 'NULL DEFAULT NULL' : ($default ? " default $default" : ''));
```

Comment:
If you have a strict check for the `default`, why not have a strict check for the `mandatory`?
Will this be correct?
```suggestion
                $fields[$field_name] = "tinyint $mandatory" . (is_null($default) && empty($mandatory) ? 'NULL DEFAULT NULL' : ($default ? " default $default" : ''));
```
If yes, use this syntax to be precise, if not, use `$mandatory === ''`